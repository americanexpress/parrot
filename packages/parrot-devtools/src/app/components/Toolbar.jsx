/*
 * Copyright 2020 American Express Travel Related Services Company, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
 * or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import React from 'react';
import PropTypes from 'prop-types';
import { SearchBar, ClearButton } from './styled';

const Toolbar = ({ scenariosData }) => {
  const { filterValue, setFilterValue } = scenariosData;
  const handleFilterChange = e => {
    scenariosData.setFilterValue(e.target.value.toLowerCase());
  };
  return (
    <div className="pad-1-b flex flex-justify-center flex-align-items-center">
      <SearchBar
        value={filterValue}
        onChange={handleFilterChange}
        placeholder="filter scenarios..."
      />
      <ClearButton className="dls-glyph-cancel-circle" onClick={() => setFilterValue('')} />
    </div>
  );
};

export default Toolbar;

Toolbar.propTypes = {
  scenariosData: PropTypes.shape({
    filterValue: PropTypes.string,
    setFilterValue: PropTypes.func,
  }).isRequired,
};
