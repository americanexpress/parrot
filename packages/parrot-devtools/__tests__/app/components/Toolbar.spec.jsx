/*
 * Copyright 2020 American Express Travel Related Services Company, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
 * or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import React from 'react';
import { shallow } from 'enzyme';

import Toolbar from '../../../src/app/components/Toolbar';

describe('Toolbar', () => {
  const setFilterValue = jest.fn();
  const mockProps = {
    filterValue: '',
    setFilterValue,
  };
  it('should render Toolbar component', () => {
    const result = shallow(<Toolbar scenariosData={{ ...mockProps }} />);

    expect(result).toMatchSnapshot();
  });

  it('invokes setFilterValue', () => {
    const result = shallow(<Toolbar scenariosData={{ ...mockProps }} />);

    const event = {
      target: { value: 'abc' },
    };
    result.find('input').simulate('change', event);
    expect(setFilterValue).toHaveBeenCalledWith('abc');
  });

  it('clear button', () => {
    const result = shallow(<Toolbar scenariosData={{ ...mockProps }} />);
    result.find('button').simulate('click');
    expect(setFilterValue).toHaveBeenCalledWith('');
  });
});
