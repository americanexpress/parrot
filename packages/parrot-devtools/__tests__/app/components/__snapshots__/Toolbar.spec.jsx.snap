// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toolbar should render Toolbar component 1`] = `
<div
  className="pad-1-b flex flex-justify-center flex-align-items-center"
>
  <input
    onChange={[Function]}
    placeholder="filter scenarios..."
    value=""
  />
  <button
    className="dls-glyph-cancel-circle"
    onClick={[Function]}
  />
</div>
`;
